alerting/src/main/kotlin/org/opensearch/alerting/chainedAlertCondition/parsers/ChainedAlertExpressionParser.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
```

Comment:
can we reuse the logic from https://github.com/opensearch-project/alerting/blob/main/alerting/src/main/kotlin/org/opensearch/alerting/triggercondition/parsers/TriggerExpressionParser.kt as it looks like lot of the logic is same?

Reply:
There are some differences in the code. WIll take a follow up on this PR to review and have a generic expression parser for both these RPN parsers